packages/web-extension/src/messaging/BackgroundMessenger.ts

```diff
@@ -60,15 +60,22 @@ export const createBackgroundMessenger = ({ logger, runtime }: MessengerDependen
      * Disconnect all existing ports and stop listening for new ones.
      */
     destroy() {
-      for (const { message$, ports$ } of channels.values()) {
-        message$.complete();
-        for (const port of ports$.value) {
-          port.disconnect();
-        }
+      // eslint-disable-next-line unicorn/no-useless-spread
+      for (const channelName of [...channels.keys()]) {
+        this.destroyChannel(channelName);
       }
       runtime.onConnect.removeListener(onConnect);
       logger.warn('[BackgroundMessenger] destroyed');
     },
+    destroyChannel(channelName: string) {
+      const channel = channels.get(channelName);
+      if (!channel) return;
+      channel.message$.complete();
+      for (const port of channel.ports$.value) {
+        port.disconnect();
+      }
+      channels.delete(channelName);
```

Comment:
This is the only part that is new, right?
And calling destroyChannel from destroy()

Reply:
Yes, now instead of destroying all channels, it will only destroy channels with relevant name